specs/latest/2.0/index.html

```diff
@@ -3292,14 +3292,20 @@ <h3><a name="BUFFER_OBJECT_BINDING">Buffer Object Binding</a></h3>
         from other buffer data also maps better to the Direct3D API.
     </div>
 
+    <h3>Preventing undefined behavior with Transform Feedback</h3>
+
     <p>
-        In addition, a buffer can not be simultaneously bound to a transform feedback buffer binding point
-        and another buffer binding point in the WebGL 2.0 API. Attempting to violate this rule generates an
-        <code>INVALID_OPERATION</code> error, and the state of the binding point will remain untouched.
+        Attempting to read from or write to a buffer which is bound to a transform feedback buffer binding
+        point, via any mechanism other than transform feedback itself (e.g. as a vertex array in a draw
+        call), fails with an INVALID OPERATION error. There is one exception to this rule: bufferData,
+        bufferSubData, getBufferSubData, and copyBufferSubData are allowed as long as transform feedback
+        is not active.
```

Comment:
You're right, I didn't think about it this way before. The language in the GL spec is a little ambiguous, but this may be the most reasonable interpretation. I will rewrite the spec language and update the tests for this.

Reply:
Specifically the way we validate this in Firefox is each buffer has these two members:
    size_t mTFBindCount;
    size_t mNonTFBindCount;
Any operation read or write to the buffer while `mTFBindCount && mNonTFBindCount` should error.

Reply:
@jdashg Do you update the counts on each buffer when VAOs or TF objects are bound/unbound, or only when the buffer is added/removed from a VAO or TF object?

Reply:
A 'background' TFO does not count towards buffer TF bindings currently, so TFO rebinds update the counts.

Reply:
Are VAOs treated the same way?